### DISCUSSION TITLE
Run on local my changes

### DISCUSSION THREAD

**Discussion by MatteoPossamai** [aws/aws-cli#8070] [Q&A] [answered]
Hi. I would really like to start contributing with code. But I have no Idea how I should be able to run the code I change on local machine. Is there any guide or explaination about how I am supposed to use the code I modified. 

For example, I made a change in one file where I simply added a print statement to see if everything worked. Now, how am I supposed to make it run, and see the print statement I wrote? 

I wrote it inside the __main__.py function, for example. Any idea or suggestion?

**Comment by RyanFitzSimmonsAK** [comment, marked as answer]
Hi @MatteoPossamai, thanks for reaching out.  In order to test changes you've made, you'll need to have [built and installed the CLI from source](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-source-install.html).  Once you do that, you should be able to make and test changes.  I'd also recommend taking a look at the [contribution guide](https://github.com/aws/aws-cli/blob/develop/CONTRIBUTING.md), since you're interested in that.  Hope that helps!

@github-actions proposed answer

**Reply by MatteoPossamai** [reply]
Ok great. Cool. Can you tell me something about how to actually add a command, and to customize user interaction with the output?

**Reply by RyanFitzSimmonsAK** [reply]
I'd recommend taking a look at the [aws s3 commands](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3/index.html), since these aren't actually mapped to a specific API operation, but are instead defined in the CLI.  The [add-model](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3/index.html) might also be worth looking at.  Could you tell me a bit more about your use case?
